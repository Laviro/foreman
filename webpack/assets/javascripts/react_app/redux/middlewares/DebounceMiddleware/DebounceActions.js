import {
  DEBOUNCE_START,
  DEBOUNCE_CLEAR,
  DEBOUNCE_STOP_INCOMING_ACTION,
} from './DebounceConstants';

export const startDebounce = ({ key, debounceID }) => ({
  type: DEBOUNCE_START,
  payload: {
    key,
    debounceID,
  },
});

export const clearDebounce = key => ({
  type: DEBOUNCE_CLEAR,
  payload: {
    key,
  },
});

export const stopIncomingAction = key => ({
  type: DEBOUNCE_STOP_INCOMING_ACTION,
  payload: {
    key,
  },
});
