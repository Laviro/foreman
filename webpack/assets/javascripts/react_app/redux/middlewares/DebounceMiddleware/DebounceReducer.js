import Immutable from 'seamless-immutable';
import { DEBOUNCE_START, DEBOUNCE_CLEAR } from './DebounceConstants';

const initialState = Immutable({});

export const reducer = (state = initialState, action) => {
  const { type, payload: { key, debounceID } = {} } = action;
  switch (type) {
    case DEBOUNCE_START:
      return state.merge({ [key]: debounceID });
    case DEBOUNCE_CLEAR:
      return state.without(key);
    default:
      return state;
  }
};

export const reducers = { debounce: reducer };

export default reducer;
