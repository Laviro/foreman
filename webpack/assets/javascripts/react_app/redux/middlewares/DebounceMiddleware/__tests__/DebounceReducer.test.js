import Immutable from 'seamless-immutable';
import { testReducerSnapshotWithFixtures } from '@theforeman/test';
import { reducer } from '../DebounceReducer';
import { startDebounce, clearDebounce } from '../DebounceActions';
import {
  key,
  debounceID,
  stateWithKey,
  initialState,
} from '../DebounceFixtures';

const fixtures = {
  'should return the initial state': initialState,
  'should handle DEBOUNCE_START': {
    action: startDebounce({ key, debounceID }),
  },
};

describe('Debounce reducer', () => {
  testReducerSnapshotWithFixtures(reducer, fixtures);

  it('Should handle DEBOUNCE_CLEAR', () => {
    const state = Immutable(stateWithKey);
    const stopAction = clearDebounce(key);

    expect(reducer(state, stopAction)).toMatchSnapshot();
  });
});
