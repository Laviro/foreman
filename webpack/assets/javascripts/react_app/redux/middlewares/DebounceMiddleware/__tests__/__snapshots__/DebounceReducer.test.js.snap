// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Debounce reducer Should handle DEBOUNCE_CLEAR 1`] = `Object {}`;

exports[`Debounce reducer should handle DEBOUNCE_START 1`] = `
Object {
  "SOME_KEY": 121,
}
`;

exports[`Debounce reducer should return the initial state 1`] = `Object {}`;
